xontrib/tests/test_uncpushd.py

```diff
@@ -0,0 +1,134 @@
+'''Tests for xontrib.uncpushd'''
+
+import pytest
+import os
+from xontrib.uncpushd import unc_pushd, unc_popd, _unc_tempDrives, _do_subproc
+
+import builtins
+from xonsh.environ import Env
+from xonsh.built_ins import load_builtins
+from xonsh.dirstack import DIRSTACK
+from xonsh.platform import ON_WINDOWS
+
+
+@pytest.fixture(scope="module")
+def wd_setup(request):
+    """
+    test fixture creates two subdirectories under test folder containing this test file, invokes each test
+    with working directory set to tests folder and DIRPATH and PWD set in builtins.__xonsh_env__
+    (as is apparently needed by xonsh/dirstack.py, revealed by a close reading of their unit tests.)
+    """
+
+    _old_pwd = os.getcwd()
+    _start_wd = os.path.dirname(os.path.abspath(__file__))
+
+    try:
+        os.mkdir( os.path.join( _start_wd, 'cdtest1'))
+    except FileExistsError:
+        pass
+    try:
+        os.mkdir( os.path.join( _start_wd, 'cdtest2'))
+    except FileExistsError:
+        pass
+
+    load_builtins()
+    _old_env = builtins.__xonsh_env__
+    builtins.__xonsh_env__ = Env(CDPATH=os.path.dirname(_start_wd), PWD=_start_wd)
+    os.chdir( _start_wd)
+
+    def fin():
+        builtins.__xonsh_env__ = _old_env
+        os.chdir(_old_pwd)
+
+    request.addfinalizer( fin)
+
+    return _start_wd
+
+@pytest.fixture(scope="module")
+def shares_setup( request):
+    """create some shares to play with on current machine"""
+
+    if not ON_WINDOWS:
+        return []
+    else:
+        original_wd = os.getcwd()
+        shares = [[r'uncpushd_test_cwd', 'y:', os.path.abspath(os.getcwd())] \
+                  , [r'uncpushd_test_cd1', 'w:', os.path.abspath(os.path.join(os.getcwd(), 'cdtest1'))]]
+        for s,d,l in shares:      # set up some shares on local machine
+            _do_subproc(['net', 'share', s + '=' + l], msg='setup')
+            _do_subproc(['net', 'use', d, r"\\localhost" + '\\' + s], msg='setup')
+
+        def fin():
+            # we want to delete the test shares we've created, but can't do that if unc shares in DIRSTACK
+            # (left over from assert fail aborted test)
+            os.chdir( original_wd)
+            for dl in _unc_tempDrives:
+                _do_subproc(['net', 'use', dl, '/delete'], msg='teardown1')
+            for s,d,l in shares:
+                _do_subproc(['net', 'use', d, '/delete'], msg='teardown2')
+                _do_subproc(['net', 'share', s, '/delete'], msg='teardown')
+
+        request.addfinalizer(fin)
+
+        return [ [ r"\\localhost" +'\\'+ s[0], s[1] , s[2]] for s in shares]
```

Comment:
Same here. You can simplify it a bit with a yield fixture. Not a blocking issue of course :)